scratchml/models/svc.py

```diff
@@ -0,0 +1,202 @@
+from scratchml.utils import split_data_into_batches
+from ..metrics import accuracy, recall, precision, f1_score, confusion_matrix
+from scratchml.regularizations import l2
+from scratchml.scalers import StandardScaler
+import numpy as np
+
+
+class SVC:
+    """
+    Support Vector Classifier (SVC) with options for linear, polynomial, and RBF kernels.
+    """
+
+    _valid_metrics = ["accuracy", "precision", "recall", "f1_score", "confusion_matrix"]
+
+    def __init__(
+        self,
+        C=0.35,
+        alpha=0.01,
+        kernel="linear",
+        degree=3,
+        max_iter=1500,
+        tol=1e-4,
+        learning_rate=1e-5,
+        decay=0.999,
+        batch_size=64,
+        early_stopping=True,
+        adaptive_lr=False,
+    ):
+        if C <= 0:
+            raise ValueError("Regularization parameter C must be positive.")
+
+        self.C = C
+        self.alpha = alpha
+        self.kernel = kernel
+        self.degree = degree
+        self.max_iter = max_iter
+        self.tol = tol
+        self.learning_rate = learning_rate
+        self.decay = decay
+        self.batch_size = batch_size
+        self.early_stopping = early_stopping
+        self.adaptive_lr = adaptive_lr
+        self.weights = None
+        self.classes_ = None
+        self.bias = 0
+        self.scaler = StandardScaler()
+        self._validate_parameters()
+
+    def _apply_kernel(self, X1, X2):
+        """Applies the selected kernel function."""
+        if self.kernel == "linear":
+            return np.dot(X1, X2.T)
+        elif self.kernel == "polynomial":
+            return (1 + np.dot(X1, X2.T)) ** self.degree
+        elif self.kernel == "rbf":
+            gamma = 1 / X1.shape[1]
+            return np.exp(
+                -gamma * np.square(np.linalg.norm(X1[:, np.newaxis] - X2, axis=2))
+            )
+
+    def _validate_parameters(self):
```

Comment:
Please, add typing here too. Like this:

```python3
def _validate_parameters(self) -> None:
```

Reply:
Please, add a docstring as well. Like this:

```python3
def _validate_parameters(self) -> None:
   """
   Auxiliary function used to validate the values of the parameters
   passed during the initialization.
   """
```

Reply:
Please, add a validation for other parameters as well (such as kernel, alpha, max_iter, degree, tol, and others). You can verify the parameter value and its type. Check out other implementations to get an idea of how to do that (e.g., for the Random Forest Classifier implementation, check line 241 in file random_forest.py).